ide-migration/server/migration/api/track-service.js

```diff
@@ -0,0 +1,131 @@
+ *
+ * SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+let dao = require('db/v4/dao');
+let database = require('db/v4/database');
+let migrationsTable = null;
+
+
+class TrackService {
+    setupTable() {
+        migrationsTable = dao.create({
+            table: "MIGRATIONS",
+            properties: [{
+                name: "id",
+                column: "ID",
+                type: "BIGINT",
+                id: true
+            }, {
+                name: "deliveryUnit",
+                column: "DELIVERY_UNIT",
+                type: "VARCHAR",
+                required: false
+            }, {
+                name: "startedOn",
+                column: "STARTED_ON",
+                type: "TIMESTAMP",
+                required: false
+            }, {
+                name: "lastUpdated",
+                column: "LAST_UPDATED",
+                type: "TIMESTAMP",
+                required: false
+            },
+                {
+                    name: "status",
+                    column: "STATUS",
+                    type: "VARCHAR",
+                    required: false
+                }]
+
+        });
+
+        if (!migrationsTable.existsTable()) {
+            migrationsTable.createTable();
+        }
+
+    }
+
+    addEntry(duName, time, status) {
+        if (this.checkDu(duName) === 0) {
+            migrationsTable.insert({
+                deliveryUnit: duName,
+                startedOn: time,
+                lastUpdated: Date.now(),
+                status: status
+            });
+        }
+    }
+
+    updateEntry(deliveryUnit) {
+        let connection = database.getConnection();
+        try {
+            let statement = connection.prepareStatement("UPDATE MIGRATIONS SET STATUS ='MIGRATED', LAST_UPDATED = CURRENT_TIMESTAMP WHERE DELIVERY_UNIT='" + deliveryUnit + "'");
+            statement.executeUpdate();
+            statement.close();
+        } catch (e) {
+            console.trace(e);
+            console.log(e.message);
+        } finally {
+            connection.close();
+        }
+    }
+
+    updateOnStart(deliveryUnit) {
+        let connection = database.getConnection();
+        try {
+            let statement = connection.prepareStatement("UPDATE MIGRATIONS SET STATUS ='PROCESSING', LAST_UPDATED = CURRENT_TIMESTAMP WHERE DELIVERY_UNIT='" + deliveryUnit + "'");
+            statement.executeUpdate();
+            statement.close();
+        } catch (e) {
+            console.trace(e);
+            console.log(e.message);
+        } finally {
+            connection.close();
+        }
+    }
+
+    updateOnFail(deliveryUnit) {
+        let connection = database.getConnection();
+        try {
+            let statement = connection.prepareStatement("UPDATE MIGRATIONS SET STATUS ='FAILED', LAST_UPDATED = CURRENT_TIMESTAMP WHERE DELIVERY_UNIT='" + deliveryUnit + "'");
```

Comment:
Use parameterized  prepare statement instead of concatinating the `deliveryUnit`